File: SYP.Web/Modules/Setting/VatRates/Editor/VatRatesEditorColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.VatRatesEditor")]
[BasedOnRow(typeof(VatRatesRow), CheckNames = true)]
public class VatRatesEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Code { get; set; }
    public bool IsDefault { get; set; }
    public bool IsActive { get; set; }
    public int SortOrder { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Setting/VatRatesEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { VatRatesRow } from "./VatRatesRow";

export interface VatRatesEditorColumns {
    Id: Column<VatRatesRow>;
    Name: Column<VatRatesRow>;
    Code: Column<VatRatesRow>;
    IsDefault: Column<VatRatesRow>;
    IsActive: Column<VatRatesRow>;
    SortOrder: Column<VatRatesRow>;
}

export class VatRatesEditorColumns extends ColumnsBase<VatRatesRow> {
    static readonly columnsKey = 'Setting.VatRatesEditor';
    static readonly Fields = fieldsProxy<VatRatesEditorColumns>();
}